### PR TITLE
Balance widget

### PR DESCRIPTION
**Pull request by abocquet**

### CODE REVIEW

File: backend/associations/views/marketplace.py [hunk 1]
```diff
@@ -271,12 +271,21 @@ def compute_balance(user, marketplace):
...
             "marketplace": marketplace.id,
             "user": user.id,
         }
 
+        try:
+            res["association"] = {
+                "name": marketplace.association.name
+            }
+        except Marketplace.association.RelatedObjectDoesNotExist:
+            pass
```
Comment (Piwakk) [inline backend/associations/views/marketplace.py:285]: How is that even possible?

File: frontend/src/components/widgets/BalanceWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import { useWidgetConfig } from "./widgetConfig";
+import { Loading } from "../utils/Loading";
+import { ErrorMessage } from "../utils/ErrorPage";
+import React from "react";
+import { formatPrice } from "../../utils/format";
+import "./widget_balance.css";
+
+export const BalanceWidget = ({ ...props }) => {
+  const { data, error, status } = useBetterQuery<any>(
+    ["subsriptions.balance.get"],
```
Comment (Piwakk) [inline frontend/src/components/widgets/BalanceWidget.tsx:13]:
```suggestion
    ["subscriptions.balance.get"],
```

File: frontend/src/components/widgets/BalanceWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        <Loading />
+      ) : status === "error" ? (
+        <ErrorMessage>`Une erreur est apparue: ${error}`</ErrorMessage>
+      ) : status === "success" && data ? (
+        <>
+          {data.every((balance) => balance.balance >= 0) ? (
+            <img
+              className={"m-auto balance-gif"}
+              src="/assets/balance/balance-positive.gif"
+              alt="All balances are positive"
```
Comment (Piwakk) [inline frontend/src/components/widgets/BalanceWidget.tsx:39]:
```suggestion
              alt="Tous les soldes sont positifs"
```

File: frontend/src/components/widgets/BalanceWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            />
+          ) : (
+            <img
+              className={"m-auto balance-gif"}
+              src="/assets/balance/balance-negative.gif"
+              alt="Not all balance are positive"
```
Comment (Piwakk) [inline frontend/src/components/widgets/BalanceWidget.tsx:45]:
```suggestion
              alt="Au moins un solde est négatif"
```

**Review by Piwakk — APPROVED** [review]

File: backend/associations/views/marketplace.py (hunk 1 above)
Comment (abocquet) [reply backend/associations/views/marketplace.py:285]: Because of the fixtures

**Pull request merged**